addon/components/validated-form.js

```diff
@@ -2,9 +2,15 @@ import { resolve } from 'rsvp';
 import { computed } from '@ember/object';
 import { getOwner } from '@ember/application';
 import Component from '@ember/component';
+import { deprecate } from '@ember/application/deprecations';
 import layout from '../templates/components/validated-form';
 
 function runTaskOrAction(taskOrAction, model) {
+  deprecate("DEPRECATED passing a task to `on-submit` is deprecated. Please replace it with `on-submit=(perform myTask)`",
+    taskOrAction.perform,
```

Comment:
I think this should be `!taskOrAction.perform`, since the deprecation warning is displayed when the condition is _falsy_

Reply:
Whoops, good catch, that's embarrassing! Fixed and rebased to hide my shame 🤦‍♂️

Reply:
Awesome! Thanks for following this through, and no worries about this one - you were only off by one bit :wink: 
I'll cut a release tomorrow!